In [1]:
from pathlib import Path
from keyword_search_evaluation import get_scores

/usr/shared/CMPT/scratch/alp11/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"Related keywords score: {scores["related_keyword_score"] * 100:.2f}%")

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 20386.56it/s]


models/vanilla-finetuned
Full match score: 43.68%
Related keywords score: 5.86%



Loading weights: 100%|██████████| 199/199 [00:01<00:00, 109.95it/s]


models/jargon-layman-finetuned-0.75
Full match score: 45.55%
Related keywords score: 6.06%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 205.70it/s]


models/jargon-layman-finetuned-1.00
Full match score: 46.05%
Related keywords score: 6.12%



Loading weights: 100%|██████████| 199/199 [00:02<00:00, 94.32it/s] 


models/jargon-layman-finetuned-0.50
Full match score: 46.05%
Related keywords score: 6.05%



Loading weights: 100%|██████████| 199/199 [00:02<00:00, 91.63it/s]


models/jargon-layman-finetuned-0.25
Full match score: 45.65%
Related keywords score: 5.99%



Loading weights: 100%|██████████| 310/310 [00:00<00:00, 18243.27it/s]
Loading weights: 0it [00:00, ?it/s]
Qwen3Model LOAD REPORT from: models/vanilla-qwen
Key                                                                     | Status     | 
------------------------------------------------------------------------+------------+-
base_model.model.layers.{0...27}.self_attn.v_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.down_proj.lora_B.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.k_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.o_proj.lora_A.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.up_proj.lora_B.default.weight      | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.q_proj.lora_A.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.gate_proj.lora_A.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.up_proj.lora_A.d

models/vanilla-qwen
Full match score: 48.42%
Related keywords score: 5.97%



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 85484.07it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


allenai/scibert_scivocab_uncased
Full match score: 24.90%
Related keywords score: 3.50%


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 11.46% | 1.94% |
| vanilla-finetuned | <u>43.68%</u> | <u>5.86%</u> |
| jargon-layman-finetuned | **45.95%** | **6.12%** |

In [6]:
import pandas as pd

In [7]:
scores_for_select_model = get_scores("models/jargon-layman-finetuned-1.00")
search_results: list[list[dict[str, float]]] = scores_for_select_model["search_results"]
search_df: pd.DataFrame = scores_for_select_model["search_df"]

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 18411.90it/s]


In [8]:
search_results

[[{'corpus_id': 0, 'score': 0.8965789079666138},
  {'corpus_id': 10, 'score': 0.8111602067947388},
  {'corpus_id': 12, 'score': 0.7716555595397949},
  {'corpus_id': 2, 'score': 0.7694507837295532},
  {'corpus_id': 4, 'score': 0.752529501914978},
  {'corpus_id': 14, 'score': 0.7389509677886963},
  {'corpus_id': 11, 'score': 0.7259838581085205},
  {'corpus_id': 1, 'score': 0.707138180732727},
  {'corpus_id': 7, 'score': 0.7052755355834961},
  {'corpus_id': 8, 'score': 0.6937223076820374},
  {'corpus_id': 6, 'score': 0.689222514629364},
  {'corpus_id': 9, 'score': 0.6870524883270264},
  {'corpus_id': 3, 'score': 0.6774637699127197},
  {'corpus_id': 5, 'score': 0.6674441695213318},
  {'corpus_id': 672, 'score': 0.6466033458709717}],
 [{'corpus_id': 1, 'score': 0.7851189374923706},
  {'corpus_id': 3, 'score': 0.7493757009506226},
  {'corpus_id': 4, 'score': 0.7332364320755005},
  {'corpus_id': 0, 'score': 0.699624240398407},
  {'corpus_id': 14, 'score': 0.6860028505325317},
  {'corpus_id': 

In [9]:
search_df

,path,jargon,layman
0,data/test_keywords/0080fed1c3048bb930467487085...,Magnetic thermoplastic nanofibers,Tiny magnetic fibers made from plastic
1,data/test_keywords/0080fed1c3048bb930467487085...,Immiscible polymer blends,Two polymers that don't mix well
2,data/test_keywords/0080fed1c3048bb930467487085...,Fe3O4 nanoparticles,Tiny magnetic particles
3,data/test_keywords/0080fed1c3048bb930467487085...,PVA-co-PE,A special plastic blend
4,data/test_keywords/0080fed1c3048bb930467487085...,Cellulose acetate butyrate matrix,A plastic base layer
...,...,...,...
1007,data/test_keywords/0f86b606933b75acd5cfde60be5...,Disappearance in 1938,Vanished without trace
1008,data/test_keywords/0f86b606933b75acd5cfde60be5...,Scientific legacy,Lasting contributions to science
1009,data/test_keywords/0f86b606933b75acd5cfde60be5...,Personal background,Private life and history
1010,data/test_keywords/0f86b606933b75acd5cfde60be5...,Recent research findings,New discoveries about his work


In [10]:
results_df = pd.DataFrame(data={
    "query_index": range(len(search_results)),
    "result_id": [
        [result["corpus_id"] for result in single_results]
        for single_results in search_results
    ]
}).explode("result_id")
results_df.head()

,query_index,result_id
0,0,0
0,0,10
0,0,12
0,0,2
0,0,4


In [11]:
results_df["layman"] = results_df["query_index"].apply(lambda i: search_df["layman"].iloc[i])
results_df["jargon"] = results_df["result_id"].apply(lambda i: search_df["jargon"].iloc[i])
results_df.head()

,query_index,result_id,layman,jargon
0,0,0,Tiny magnetic fibers made from plastic,Magnetic thermoplastic nanofibers
0,0,10,Tiny magnetic fibers made from plastic,Uniform dispersion of Fe3O4 nanoparticles
0,0,12,Tiny magnetic fibers made from plastic,Saturation magnetization
0,0,2,Tiny magnetic fibers made from plastic,Fe3O4 nanoparticles
0,0,4,Tiny magnetic fibers made from plastic,Cellulose acetate butyrate matrix


In [12]:
search_results[0][0]

{'corpus_id': 0, 'score': 0.8965789079666138}